# S06: Chi-Square



## Categorical Data Analysis with $\chi^2$

The goals of <span style = 'color:blue; font-weight:bold'>linear regression</span> and the <span style = 'color:blue; font-weight:bold'>$\chi^2$ test of independence</span> are very similar:

- A **linear regression** determines if a relationship exists between 2 <span style = 'color:red; font-weight:bold'>numeric</span> variables.
- A **$\chi^2$ test of independence** determines if a relationship exists between 2 <span style = 'color:red; font-weight:bold'>category</span> variables.

## Data Verification

To use $\chi^2$ Test of Independence, we must have no more than 20\% low cell counts in the **expected matrix** where "low cell count" is defined to be **less than five.**

The examples below begin with the **observed data matrix** being constructed directly, while the third example will need to calculate the values in the data matrix using **xtabs()**.

## Example 1: Learning Styles

A psychologist questions whether preference of learning method differs by gender. A group survey is conducted the results of which are shown below. Test whether a relationship exists between gender and learning method.

<table style="width:40%">
  <tr>
    <th></th>
    <th style="text-align: center">Male</th>
    <th style="text-align: center">Female</th> 
  </tr>
  <tr>
    <td style="text-align: center"><b>Visual</b></td>
    <td style="text-align: center">25</td>
    <td style="text-align: center">14</td>
  </tr>
  <tr>
    <td style="text-align: center"><b>Auditory</b></td>
    <td style="text-align: center">24</td>
    <td style="text-align: center">42</td>
  </tr>
  <tr>
    <td style="text-align: center"><b>Kinethetic</b></td>
    <td style="text-align: center">41</td>
    <td style="text-align: center">34</td>
  </tr>
</table>

In [4]:
data <- matrix(c(25, 24, 41, 14, 42, 34), nrow = 3)
data

25,14
24,42
41,34


In [5]:
result <- chisq.test(data)
result$expected

19.5,19.5
33.0,33.0
37.5,37.5


**Data Check**

No values in the expected matrix are less than 5, so these data are appropriate for $\chi^2$ procedures.

**Hypotheis**

- $H_0 : $ Proportions of learning styles are **independent** of biological sex.
- $H_a : $ Proportions of learning styles are **dependent** upon biological sex.

In [6]:
result


	Pearson's Chi-squared test

data:  data
X-squared = 8.665, df = 2, p-value = 0.01313


**Reporting Out**

Because $p = 0.0.1313 < 0.05 = \alpha$, we reject the null.  Evidence suggests that the proportion of learning styles depends upon biological sex.

## Example 2: Snacks vs. Type of Movie

Conduct a test of indepence upon Type of Movie and the proportion of attendees who purchased snacks to take into the movie. Test at the $\alpha = 0.05$ level.

<table style="width:35%">
  <tr>
    <th></th>
    <th style="text-align: center">Snacks</th>
    <th style="text-align: center">No Snacks</th> 
  </tr>
  <tr>
    <td style="text-align: center"><b>Action</b></td>
    <td style="text-align: center">55</td>
    <td style="text-align: center">75</td>
  </tr>
  <tr>
    <td style="text-align: center"><b>Comedy</b></td>
    <td style="text-align: center">135</td>
    <td style="text-align: center">155</td>
  </tr>
  <tr>
    <td style="text-align: center"><b>Family</b></td>
    <td style="text-align: center">85</td>
    <td style="text-align: center">55</td>
  </tr>
  <tr>
    <td style="text-align: center"><b>Horror</b></td>
    <td style="text-align: center">45</td>
    <td style="text-align: center">15</td>
  </tr>
</table>

In [7]:
obs <- matrix(c(55, 135, 85, 45, 75, 155, 55, 15), nrow = 4)
obs

55,75
135,155
85,55
45,15


In [8]:
res <- chisq.test(obs)
res$expected

67.09677,62.90323
149.67742,140.32258
72.25806,67.74194
30.96774,29.03226


**Data Check**

No values in the expected matrix are less than 5, so these data are appropriate for $\chi^2$ procedures.

**Hypothesis**

- $H_0 : $ Perecentage of moviegoers buying snacks is **independent** of type of movie seen.
- $H_a : $ Perecentage of moviegoers buying snacks is **dependent** upon type of movie seen.

In [11]:
res


	Pearson's Chi-squared test

data:  obs
X-squared = 25.266, df = 3, p-value = 1.358e-05


**Reporting out**

Since $p = 1.358\times 10^{-5} < 0.05 =\alpha$, we reject the null. Thus, evidence suggests that whether or not a patron buys snacks depends upon the type of movie they are going to see.

## Example 3: Accept the Date

Using the **AccDate** varible in the **personality** data set, test at the $0.05$ level of significance whether Yes/No responses to the dating question depends upon biological sex.

In [14]:
pers <- read.csv('https://faculty.ung.edu/rsinn/data/personality.csv')

In [20]:
xtabs(~Sex + AccDate, data = pers)

   AccDate
Sex  N  Y
  F 28 46
  M 28 27

In [21]:
obs2 <- matrix(c(28,28,46,27), nrow = 2)
obs2

28,46
28,27


In [23]:
rz <- chisq.test(obs2)
rz$expected

32.12403,41.87597
23.87597,31.12403


**Data Checks**

No observed matrix values less than 5. OK.

**Hypothesis**

- $H_0 : $ Vars independent
- $H_a : $ Vars dependent

In [25]:
rz


	Pearson's Chi-squared test with Yates' continuity correction

data:  obs2
X-squared = 1.6945, df = 1, p-value = 0.193


Since $p= 0.193 > 0.05 = \alpha$, we fail to reject the null. We thus have no evidence that the Yes/No responses depend upon biological sex.